asyncscope.md

```diff
@@ -318,84 +321,145 @@ struct async_scope {
     async_scope& operator=(const async_scope&) = delete;
     async_scope& operator=(async_scope&&) = delete;
 
-    auto nest(sender) const&->nested_sender<Values…>;
-
-    auto spawn(sender) const&->void;
-    auto spawn_future(sender) const&->future_sender<Values…>;
+    template <sender S>
+    void spawn(S&&);
 
-    auto on_empty() const&->empty_sender;
+    template <sender S>
+    @_spawn-future-sender_@<S> spawn_future(S&&);
 
-    auto get_stop_source() & ->stop_source&;
-    auto get_stop_token() const& ->stop_token;
-    auto request_stop() & ->void;
-};
+    template <sender S>
+    @_nest-sender_@<S> nest(S&&);
 
-template<class... Ts>
-struct nested_sender; @_see below_@
+    @_on-empty-sender_@<S> on_empty() const noexcept;
 
-template<class... Ts>
-struct future_sender; @_see below_@
+    in_place_stop_source& get_stop_source() noexcept;
+    in_place_stop_token get_stop_token() const noexcept;
+    void request_stop() noexcept;
+};
 
-struct empty_sender; @_see below_@
+template<typename S>
+struct @_nest-sender_@; // @_see below_@
+template<typename S>
+struct @_spawn-future-sender_@; // @_see below_@
+template<typename S>
+struct @_on-empty-sender_@; // @_see below_@
 ```
 
 ## Lifetime
 
-An `async_scope` object must outlive work that is spawned on it. It should be viewed as owning the storage for that work.
+An `async_scope` object must outlive work that is spawned on it.
```

Comment:
Note: everywhere I can, I split paragraphs into multiple lines, so that we have one sentence per line. This makes the diffing much easier with a lot of tools.

Please excuse me if there are too many changes for things that shouldn't be